main.go

```diff
@@ -732,9 +734,37 @@ func executeLinter(state *linterState, args []string) error {
 			state.Name)
 	}
 
-	processOutput(state, buf.Bytes())
+	cmdOutput, err := preprocessLinterPanic(state.Name, buf, err)
+	processOutput(state, cmdOutput)
 	debug("%s linter took %s", state.Name, time.Since(start))
-	return nil
+	return err
+}
+
+const panicExitStatus = 2
+
+func preprocessLinterPanic(name string, buf *bytes.Buffer, processError error) ([]byte, error) {
+	if exitStatus(processError) == panicExitStatus {
```

Comment:
I think `exitStatus` should instead be `panicExitStatus` and take the approach "Returns true if error is an *os.ExitError with an exit status of 2, else returns false". Then this function can return the same output it was provided if it didn't positively. Currently `exitStatus`  

Although there's no gurantee the exit status of 2 is a panic, other applications could exit with the code 2 and go itself could (there's not docs to gurantee) exit a panic with a different code. Perhaps we could just check if panic if the error is non-nil and remove `exitStatus` completely and only call `preprocessLinterPanic` if error is non-zero.


```go

    if err != nil && linterProbablyPaniced(buf.Bytes()) {
            return fmt.Errorf("linter %s may have panicked, check output with --debug", state.Name)
    }
    processOutput(state, buf.Bytes())
    debug("%s linter took %s", state.Name, time.Since(start))
    return nil
}

func preprocessLinterPanic(buf *bytes.Buffer) bool {
    routines, _ := stack.ParseDump(buf, ioutil.Discard)
    return len(routines) > 0
}
```

Even then, perhaps this shouldn't return an error, and instead log a warning?